### DISCUSSION TITLE
v2.9.58

### DISCUSSION THREAD

**Discussion by fiftin** [semaphoreui/semaphore#1867] [Announcements]
### Features

* Integrations. Allows to integrate Semaphore with external services, like GitHub actions and GitLab (https://github.com/ansible-semaphore/semaphore/pull/1365 by @andreas-marschke):  <br><br><img width="500" alt="image" src="https://github.com/ansible-semaphore/semaphore/assets/914224/01500433-8ea3-4af1-9d33-18683bd65516">
* Rocket Chat notifications (https://github.com/ansible-semaphore/semaphore/pull/1836 by @oussjarrousse)
* Hide/show password button by @ewatkins.

### Bugfixes
* A lot to fixes and refactoring by @tboerger
* Integration authentication fixed. Following variants now available:
  * HMAC
  * GItHub Webhooks (like HMAC but with GitHub specific)
  * Token (useful for GitLab)
  * None

Integrations documenatition: https://docs.semui.co/user-guide/integrations

<hr /><em>This discussion was created from the release <a href='https://github.com/ansible-semaphore/semaphore/releases/tag/v2.9.58'>v2.9.58</a>.</em>

**Comment by fanfuvac** [comment]
Can anyone elaborate on:
Support complex expressions for OpenID claims. For example you can use following expression for email_claim:
{ "email_claim": "email | {{ .id }}@mydomain.com" }

How do I derive a name form email, if the email has format givenname.lastname@company.com ? Can I somehow create name claim which will take email and returned givenname lastname from the email address?
